coiote-aws-iot-cloud-formation/cloudFormation.json

```diff
@@ -0,0 +1,251 @@
+{
+  "AWSTemplateFormatVersion": "2010-09-09",
+  "Resources": {
+    "IamForLambda": {
+      "Type": "AWS::IAM::Role",
+      "Properties": {
+        "AssumeRolePolicyDocument": {
+          "Version": "2012-10-17",
+          "Statement": [
+            {
+              "Action": "sts:AssumeRole",
+              "Principal": {
+                "Service": "lambda.amazonaws.com"
+              },
+              "Effect": "Allow"
+            }
+          ]
+        },
+        "RoleName": "IamForLambda",
+        "ManagedPolicyArns": [
+          "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+        ]
+      }
+    },
+    "IamForOperationRequest": {
+      "Type": "AWS::IAM::Role",
+      "Properties": {
+        "AssumeRolePolicyDocument": {
+          "Version": "2012-10-17",
+          "Statement": [
+            {
+              "Action": "sts:AssumeRole",
+              "Principal": {
+                "Service": "iot.amazonaws.com"
+              },
+              "Effect": "Allow"
+            }
+          ]
+        },
+        "RoleName": "IamForOperationRequest",
+        "Policies": [
+          {
+            "PolicyName": "OperationResponseCloudWatch",
+            "PolicyDocument": {
+              "Version": "2012-10-17",
+              "Statement": [
+                {
+                  "Sid": "AllowErrorOrMessageLogWriting",
+                  "Effect": "Allow",
+                  "Action": [
+                    "logs:PutLogEvents",
+                    "logs:CreateLogStream"
+                  ],
+                  "Resource": "*"
+                }
+              ]
+            }
+          }
+        ]
+      }
+    },
+    "IamForOperationResponse": {
+      "Type": "AWS::IAM::Role",
+      "Properties": {
+        "AssumeRolePolicyDocument": {
+          "Version": "2012-10-17",
+          "Statement": [
+            {
+              "Action": "sts:AssumeRole",
+              "Principal": {
+                "Service": "iot.amazonaws.com"
+              },
+              "Effect": "Allow"
+            }
+          ]
+        },
+        "RoleName": "IamForOperationResponse",
+        "Policies": [
+          {
+            "PolicyName": "OperationResponseRepublish",
+            "PolicyDocument": {
+              "Version": "2012-10-17",
+              "Statement": [
+                {
+                  "Sid": "IotPublish",
+                  "Effect": "Allow",
+                  "Action": "iot:Publish",
+                  "Resource": "*"
```

Comment:
The same suggestion here, please try:
```
arn:aws:iot:*:*:topic/*
```
Or maybe you can even use 
```
!Ref "AWS::Region"
```
as described here https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/pseudo-parameter-reference.html and add to these roles only permissions that are needed rather than starting with a *. 
 
In our terraform version the Resources was set like here: 
https://github.com/Passarinho4/terraform-aws-iot-topic-rule/blob/master/locals.tf